core/meterer/accountledger.go

```diff
@@ -0,0 +1,364 @@
+	//     - "quorum X not enabled for on-demand": Requested quorum doesn't support on-demand
+	//     - "insufficient ondemand payment": Would exceed available balance
+	//     - "quorum config not found for X": Missing payment configuration
+	RecordOnDemandUsage(
+		ctx context.Context,
+		accountID gethcommon.Address,
+		numSymbols uint64,
+		quorumNumbers []core.QuorumID,
+		params *PaymentVaultParams,
+	) (*big.Int, error)
+
+	// SetAccountState updates the ledger with complete account state including
+	// on-chain settings (reservations, on-demand) and local tracking data.
+	//
+	// This method completely replaces the current ledger state with the provided state.
+	// It performs deep copying of all data to ensure isolation between the provided
+	// state and the internal ledger state.
+	//
+	// Use Cases:
+	//   - Initial setup: Loading account state from on-chain data
+	//   - Synchronization: Updating local state with fresh on-chain data
+	//   - Testing: Setting up known state for test scenarios
+	//
+	// Parameters:
+	//   state: The complete account state to set. Nil fields will be initialized
+	//          with appropriate zero values (empty maps, zero big.Int, etc.)
+	//
+	// Thread Safety:
+	// This method is safe for concurrent use. It acquires an exclusive write lock
+	// for the entire operation to ensure atomic state replacement.
+	SetAccountState(state AccountState)
+
+	// GetAccountState returns the complete current account state including on-chain
+	// settings and local tracking data for inspection and synchronization.
+	//
+	// The returned state is a deep copy of the internal ledger state, ensuring that
+	// modifications to the returned data will not affect the ledger's internal state.
+	// This prevents accidental mutations while allowing safe inspection and analysis.
+	//
+	// Use Cases:
+	//   - State inspection: Examining current payment usage and limits
+	//   - Synchronization: Extracting state for storage or transmission
+	//   - Testing: Verifying state after operations
+	//   - Debugging: Analyzing account state for troubleshooting
+	//
+	// Returns:
+	//   AccountState: A deep copy of the current account state including:
+	//     - All reservation settings and their current usage
+	//     - On-demand payment balance and cumulative usage
+	//     - Complete period records history
+	GetAccountState() AccountState
+}
+
+// AccountState represents the complete payment state for an account,
+// including both on-chain settings and local usage tracking.
+//
+// This struct serves as the canonical representation of all payment-related
+// data for a single account, enabling atomic state transitions and consistent
+// snapshots for synchronization between different system components.
+//
+// State Categories:
+//  1. On-chain Settings: Immutable data from blockchain (reservations, on-demand)
+//  2. Local Tracking: Mutable usage counters maintained locally (period records, cumulative payment)
+type AccountState struct {
```

Comment:
IMO it helps keep file sizes more digestible if structs and interfaces are given their own file

Reply:
put in separate files [a5581f8](https://github.com/Layr-Labs/eigenda/pull/1694/commits/a5581f871327b47cc67b1a5d670962a9dcc4fe3f)

Reply:
I decided to remove the LocalAccountLedger from this PR so the only thing it contains now is the `AccountLedger` interface definition. Hopefully the review is easier this way. 
The LocalAccountLedger is split to https://github.com/Layr-Labs/eigenda/pull/1712